end-to-end_sales-forecasting/python_scripts/sales_forecasting_1_create_project_dev_preds_FI.py

```diff
@@ -0,0 +1,208 @@
+    if mapped not in FEATURE_MAP:
+        FEATURE_MAP[mapped] = []
+    FEATURE_MAP[mapped].append(post)
+    
+#save feature map for future use
+with open('feature_map.pkl', 'wb') as f:
+    pickle.dump(FEATURE_MAP, f)
+
+data_collection_name='OJ Sales Data'
+tru.add_data_collection(data_collection_name, pre_to_post_feature_map=FEATURE_MAP, provide_transform_with_model=False)
+
+data_df, column_spec = merge_dataframes_and_create_column_spec(
+                        id_col_name='index',
+                        timestamp_col_name='datetime',
+                        pre_data=X_train_pre,
+                        post_data=X_train_post,
+                        labels=y_df)
+
+tru.add_data(
+        data_split_name='training data',
+        data=data_df,
+        column_spec=column_spec)
+
+X_val_pre = pd.read_csv('./split_sim/pre_split_2023-08-20.csv',index_col=0).reset_index()
+X_val_post = pd.read_csv('./split_sim/post_split_2023-08-20.csv',index_col=0).reset_index()
+y_val = pd.read_csv('./split_sim/label_2023-08-20.csv',index_col=0).reset_index()
+#extra_val = pd.read_csv('./split_sim/extra_1.csv', index_col='datetime')
+
+val_data_df, column_spec = merge_dataframes_and_create_column_spec(
+                        id_col_name='index',
+                        timestamp_col_name='datetime',
+                        pre_data=X_val_pre,
+                        post_data=X_val_post,
+                        labels=y_val)
+
+tru.add_data(
+        data_split_name='validation data',
+        data=val_data_df,
+        column_spec=column_spec)
+
+lin_reg = pickle.load(open("linreg.pkl", 'rb'))
+model_name = 'Ridge Regression'
+tru.add_python_model(model_name, lin_reg)
+
+
+# ## **Monitoring requirement: add new data collection for second model**
+# - 1:1 dc:model req for monitoring
+# - duplicated development data into 2nd data collection
+#   - this is required to have access to FIs for all 6 model-dev_split combinations (see next section)
+tru.add_data_collection("OJ Sales Data RF", pre_to_post_feature_map=FEATURE_MAP, provide_transform_with_model=False)
+
+random_forest = pickle.load(open("rf.pkl", 'rb'))
+model_name = 'Random Forest Regressor'
+tru.add_python_model(model_name, random_forest)
+
+data_df, column_spec = merge_dataframes_and_create_column_spec(
+                        id_col_name='index',
+                        timestamp_col_name='datetime',
+                        pre_data=X_train_pre,
+                        post_data=X_train_post,
+                        labels=y_df)
+
+tru.add_data(
+        data_split_name='training data',
+        data=data_df,
+        column_spec=column_spec)
+
+tru.add_data(
+        data_split_name='validation data',
+        data=val_data_df,
+        column_spec=column_spec)
+
+# # Compute & Upload dev split Feature Influences using TruEra QII
+    
+tru.set_model("Ridge Regression")
+tru.set_data_split("training data")
+lr_train_feat_infs = tru.compute_feature_influences(stop=len(X_train_pre))
```

Comment:
Are the influences generated on the pre data or post data?

Reply:
Does diagnostics show the post data features in the features page?

Reply:
I probably should have used len(X_train_post) -- that is the data that is model executable / is used, under the covers, for the .compute execution. In practice it makes no difference, as they're both the same length. But will change/update.

There is a feature map implemented in this workflow, so the features page / all results will be based on pre-transform. In this case, this basically just amounts to showing categorical vars with multiple levels, versus the total set of binary responses of M categorics x N(M) levels.